### ISSUE TITLE
can't use joins in has_one scope block?

### ISSUE THREAD

**Issue by InteNs** [rails/rails#32196]
Trying to use a scope with `joins()` doesn't include that join in the generated sql, be it directly, via named_scopes or via default_scopes.

Below is a minimal example that reproduces this issue.
This is causing trouble because i need to scope the primary_marking based on an attribute from it's marking_code (there would be a `.where(...)` following the join

### Steps to reproduce
```rb
class DamageIncident < ApplicationRecord
  has_one :primary_marking, -> { joins(:marking_code) }
end

class PrimaryMarking < ApplicationRecord
  belongs_to :marking_code
end

irb> DamageIncident.joins(:primary_marking)
```

### Expected behavior
```sql
SELECT damage_incidents.*
FROM damage_incidents
  INNER JOIN primary_markings
    ON primary_markings.damage_incident_id = damage_incident.id
  INNER JOIN marking_codes
    ON primary_markings.marking_code_id = marking_codes.id
```

### Actual behavior
```sql
SELECT damage_incidents.*
FROM damage_incidents
  INNER JOIN primary_markings
    ON primary_markings.damage_incident_id = damage_incident.id
```

### System configuration
**Rails version**: Rails 5.1.4

**Ruby version**: ruby 2.4.1p111 (2017-03-22 revision 58053) [x86_64-darwin16]

**Comment by kamipo** [comment]
This should be fixed by #29413.
Can you try to reproduce it on master?

**Comment by InteNs** [comment]
confirmed fixed on latest rails version!
